lib/model/model.go

```diff
@@ -1234,7 +1234,7 @@ func sendIndexTo(initial bool, minLocalVer int64, conn protocol.Connection, fold
 	return maxLocalVer, err
 }
 
-func (m *Model) updateLocals(folder string, fs []protocol.FileInfo) {
+func (m *Model) updateLocals(folder string, fs []protocol.FileInfo, remoteUpdate bool) {
```

Comment:
The passing of a bool here is fine, but every place that calls updateLocals sets it using a literal (`true` or `false`, it never varies). It's also not super clear what the bool means at the call site. I suggest adding two simple wrapper methods:

```go
func (m *Model) updateLocalsFromScanning(folder string, fs []protocol.FileInfo) {
    m.updateLocals(folder, fs, true)
}

func (m *Model) updateLocalsFromPulling(folder string, fs []protocol.FileInfo) {
    m.updateLocals(folder, fs, false)
}
```

Then everything else calls these, and they are placed just above `updateLocals` in the file so when reading that part it is obvious what the bool is about. I've changed the sense of the bool as well to mean "due to scanning" instead of "due to remote update", which will make sense below.